worker/api.py

```diff
@@ -0,0 +1,70 @@
+import os
+import tempfile
+import json
+
+from fastapi import FastAPI, UploadFile, File, Form, HTTPException
+from fastapi.responses import StreamingResponse
+from ultralytics import YOLO
+
+from communication.sam2_client import Sam2Client
+from communication.openpose_client import OpenposeClient
+from masking.sam2_pose_masker import Sam2PoseMasker
+from cli import process_video
+
+app = FastAPI()
+
+# Constants
+WORKER_SAM2_BASE_PATH = 'http://sam2:8000/sam2'
+WORKER_OPENPOSE_BASE_PATH = 'http://openpose:8000/openpose'
+
+@app.post("/mask-video")
+async def mask_video(
+    options: str = Form(...),
+    video: UploadFile = File(...)
+):
+    try:
+        options = json.loads(options)
+        hiding_strategy = options.get("hiding_strategy")
+        overlay_strategy = options.get("overlay_strategy")
+
+        valid_hiding_strategies = ['solid_fill', 'transparent_fill', 'blurring', 'pixelation', 'contours', 'none']
+        valid_overlay_strategies = ['mp_hand', 'mp_face', 'mp_pose', 'none', 'openpose', 'openpose_body25b',
+                                 'openpose_face', 'openpose_body_135']
+
+        if hiding_strategy not in valid_hiding_strategies:
+            raise ValueError(f"Invalid hiding_strategy: {hiding_strategy}. Must be one of {valid_hiding_strategies}")
+        if overlay_strategy not in valid_overlay_strategies:
+            raise ValueError(f"Invalid overlay_strategy: {overlay_strategy}. Must be one of {valid_overlay_strategies}")
+    except Exception as e:
+        raise HTTPException(
+            status_code=400,
+            detail=str(e)
+        )
+
+    sam2_client = Sam2Client(WORKER_SAM2_BASE_PATH)
+    openpose_client = OpenposeClient(WORKER_OPENPOSE_BASE_PATH)
+
+    with tempfile.NamedTemporaryFile(delete=False, suffix=".mp4") as input_tmp, \
+         tempfile.NamedTemporaryFile(delete=False, suffix=".mp4") as output_tmp:
+        input_tmp.write(await video.read())
+        input_tmp_path = input_tmp.name
+        output_tmp_path = output_tmp.name
+
+    try:
+        process_video(
+            input_file=input_tmp_path,
+            output_file=output_tmp_path,
+            sam2_client=sam2_client,
+            openpose_client=openpose_client,
+            hiding_strategy=hiding_strategy,
+            overlay_strategy=overlay_strategy
+        )
+
+        return StreamingResponse(
+            open(output_tmp_path, "rb"),
+            media_type="video/mp4",
+            headers={"Content-Disposition": "attachment; filename=processed.mp4"}
+        )
+    finally:
+        os.remove(input_tmp_path)
+        os.remove(output_tmp_path)
```

Comment:
Very nice with the finally here to ensure that the files are removed. Maybe you could even move the tempfile stuff into the try block to be extra sure. But just a tiny nit

Reply:
Good idea! I just had to initialize them to None before, so that these variables exist in case the temporary file creation fails and the finally block still tries to remove the variables.